enhancements/20240604-lvm-csi-driver.md

```diff
@@ -0,0 +1,77 @@
+# LVM CSI Driver
+
+## Summary
+
+In some specific scenarios, we need better performance and could use a strict location for the volume. The LVM could be a good choice because it is a thin layer on top of the block devices. Also, it provides useful features like volume expansion and snapshots. We want to introduce the LVM CSI driver to support the LVM volume in Harvester. 
+
+### Related Issues
+
+https://github.com/harvester/harvester/issues/5724
+
+## Motivation
+
+We do not have any current solution to support local volumes, which means directly using the local disk as the volume. The LVM CSI driver could help leverage the local disk as the volume. If the user could use the strict location VM, it would help us improve performance and latency.
+
+### Goals
+
+Introduce the LVM CSI driver to support the LVM volume in Harvester.
+
+### Non-goals [optional]
+
+Support volume expansion and snapshot for the LVM volume in the first version.
+
+## Proposal
+
+We would like to start with the existing LVM CSI driver and introduce it to the Harvester. We need to make some changes to make it more suitable for the Harvester. After that, we could use the LVM CSI driver to create the LVM volume in Harvester.
+
+### User Stories
+
+#### The user creates the PCI pass-through VM and wants a higher performance.
+When users use the vGPU feature with PCI pass-through, it means they will redirect the VM to a specific host. Also, the user wants a higher-performance disk for the VM. The LVM CSI driver could help users create the PVC with an LVM volume on a local disk. That could help users have better performance and latency.
+
+### User Experience In Detail
+
+The user creates a VM and starts with the vGPU feature. Then, the user creates the PVC with the LVM volume. The user attaches the PVC to the VM through hotplug and uses the LVM volume in the VM.
```

Comment:
we can remove the vGPU here, the feature is not restricted to vGPU VMs.

Reply:
updated!